app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerTest.kt

```diff
@@ -920,4 +853,47 @@ class BloodPressureEntrySheetControllerTest {
     verify(sheet).showBpEntryScreen()
     verifyNoMoreInteractions(sheet)
   }
+
+  @Test
+  fun `when done button is clicked in new BP entry, then save BP with entered date immediately`() {
+    val systolic = 120.toString()
+    val diastolic = 110.toString()
+    val inputDate = LocalDate.of(2016, 5, 10)
+
+    whenever(bpValidator.validate(systolic, diastolic)).thenReturn(Success(systolic.toInt(), diastolic.toInt()))
+    whenever(dateValidator.validate2(any(), any())).thenReturn(Valid(inputDate)) // TODO Use actual values
+    whenever(bloodPressureRepository.saveMeasurement(any(), any(), any(), any(), any(), any()))
+        .thenReturn(Single.just(PatientMocker.bp(patientUuid = patientUuid)))
+    whenever(appointmentRepository.markAppointmentsCreatedBeforeTodayAsVisited(patientUuid)).thenReturn(Completable.complete())
+    whenever(patientRepository.compareAndUpdateRecordedAt(any(), any())).thenReturn(Completable.complete())
+
+    with(uiEvents) {
+      onNext(BloodPressureEntrySheetCreated(OpenAs.New(patientUuid)))
+      onNext(BloodPressureScreenChanged(BP_ENTRY))
+      onNext(BloodPressureSystolicTextChanged(systolic))
+      onNext(BloodPressureDiastolicTextChanged(diastolic))
+      onNext(BloodPressureDayChanged("1"))
+      onNext(BloodPressureMonthChanged("02"))
+      onNext(BloodPressureYearChanged("16"))
+
+      reset(sheet)
+      onNext(BloodPressureSaveClicked)
+    }
+
+    val entryDateAsInstant = inputDate.atTime(OffsetTime.now(testUtcClock)).toInstant()
```

Comment:
We are using `UserClock` in the `BloodPressureEntrySheetController`, hence change this method to use the `testUserClock` to convert to `Instant`. Like this:

`inputDate.toUtcInstant(testUserClock)`